src/coreComponents/physicsSolvers/multiphysics/PoromechanicsSolver.hpp

```diff
@@ -0,0 +1,229 @@
+    array1d< real64 > r2 = addTwoVecs( s2_tilde, s1, -1.0 );
+
+    // diff = r2 - r1
+    array1d< real64 > diff = addTwoVecs( r2, r1, -1.0 );
+
+    const real64 denom = dotTwoVecs( diff, diff );
+    const real64 numer = dotTwoVecs( r1, diff );
+
+    real64 omega1 = 1.0;
+    if( !isZero( denom ))
+    {
+      omega1 = -1.0 * omega0 * numer / denom;
+    }
+    return omega1;
+  }
+
+  array1d< real64 > computeUpdate( array1d< real64 > const & s1,
+                                   array1d< real64 > const & s2_tilde,
+                                   real64 const omega1 )
+  {
+    return addTwoVecs( scalarMultiplyAVec( s1, 1.0 - omega1 ),
+                       scalarMultiplyAVec( s2_tilde, omega1 ),
+                       1.0 );
+  }
+
+  void startSequentialIteration( integer const & iter,
+                                 DomainPartition & domain ) override
+  {
+    if( m_useNA )
+    {
+      if( iter == 0 )
+      {
+        recordAverageMeanTotalStressIncrement( domain, m_s1 );
+      }
+      else
+      {
+        m_s0 = m_s1;
+        m_s1 = m_s2;
+        m_s1_tilde = m_s2_tilde;
+        m_omega0 = m_omega1;
+      }
+    }
+  }
+
+  void finishSequentialIteration( integer const & iter,
+                                  DomainPartition & domain ) override
+  {
+    if( m_useNA )
+    {
+      if( iter == 0 )
+      {
+        m_s2 = m_s2_tilde;
+        m_omega1 = 1.0;
+      }
+      else
+      {
+        m_omega1 = computeAitkenRelaxationFactor( m_s0, m_s1, m_s1_tilde, m_s2_tilde, m_omega0 );
+        m_s2 = computeUpdate( m_s1, m_s2_tilde, m_omega1 );
+        applyAcceleratedAverageMeanTotalStressIncrement( domain, m_s2 );
+      }
+    }
+  }
+
+  virtual void mapSolutionBetweenSolvers( DomainPartition & domain, integer const solverType ) override
+  {
+    if( m_useNA && solverType == static_cast< integer >( SolverType::SolidMechanics ) )
+    {
+      recordAverageMeanTotalStressIncrement( domain, m_s2_tilde );
+    }
+  }
+
+  virtual void
+  postProcessInput() override
+  {
+    Base::postProcessInput();
+
+    if( m_useNA && MpiWrapper::commSize( MPI_COMM_GEOSX ) > 1 )
+    {
+      GEOS_ERROR( "Nonlinear acceleration is not implemented for MPI runs" );
+    }
+  }
+
+  /// Member variables needed for Nonlinear Acceleration ( Aitken ). Naming convention follows ( Jiang & Tchelepi, 2019 )
+  integer m_useNA;
+  array1d< real64 > m_s0; // Accelerated averageMeanTotalStresIncrement @ outer iteration v ( two iterations ago )
+  array1d< real64 > m_s1; // Accelerated averageMeanTotalStresIncrement @ outer iteration v + 1 ( previous iteration )
+  array1d< real64 > m_s1_tilde; // Unaccelerated averageMeanTotalStresIncrement @ outer iteration v + 1 ( previous iteration )
+  array1d< real64 > m_s2; // Accelerated averageMeanTotalStresIncrement @ outer iteration v + 2 ( current iteration )
+  array1d< real64 > m_s2_tilde; // Unaccelerated averageMeanTotalStresIncrement @ outer iteration v + 1 ( current iteration )
```

Comment:
Would you consider using linear algebra vector types (`ParallelVector`) to implement this? This way you don't need to re-implement basic BLAS routines (axpy, dot), and you get MPI parallelism for free. You can still get an `ArrayView` out of them to copy the local vector potion to/from porous solid model.

The only thing you can't do in that case is use `emplace_back` (parallel vector needs to be correctly pre-sized), so you will need a simple loop over regions to calculate the total vector size beforehand, and then in the actual copy loop keep track of the starting offset in the vector for each subregion. You should also be able to use `std::copy` to move the data in/out of vector efficiently, or alternatively make the copy loops parallel.

Reply:
this is an excellent suggestion, thank you very much
will definitely work on that in the next iteration, just don't have much time to spend on that at the moment

Reply:
did the functions renaming